core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -1281,6 +1285,57 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
     assert(master.getLocations("item").isEmpty)
   }
 
+  test("SPARK-20640: Shuffle registration timeout and maxAttempts conf are working") {
+    val shufflePort = 10000
```

Comment:
I'm afraid that this may lead to flakiness in Jenkins: we run multiple concurrent builds on the machine and they aren't containerized, so hardcoding ports in unit tests risks port conflicts (especially when several jobs kick off at about the same time; this actually _is_ an issue in practice).

If you need to know the port that it binds to then I would recommend using `Utils.startServiceOnPort` (see examples of this in existing tests elsewhere in the codebase, such as in the Kafka module).